### ISSUE TITLE
'package:flutter/src/cupertino/slider.dart': Failed assertion: line 330 pos 15: 'value != null && value >= 0.0 && value <= 1.0': is not true.

### ISSUE THREAD

**Issue by LeLoMAu** [flutter/flutter#67124]
Hi all,

in Flutter I'm using Slider.adaptive in this way:

```
Slider.adaptive(
   min: 0,
   max: _songDurationInMilliseconds.toDouble(),
   value: _songDurationInMilliseconds > _currentDurationInMilliseconds ? 
      _currentDurationInMilliseconds.toDouble()
      : _songDurationInMilliseconds.toDouble(),
   onChangeStart: (double value) => audioProvider.invertSeekingState(),
   onChanged: (double value) {
      final Duration _duration = Duration(milliseconds: value.toInt(),);
      audioProvider.updatePosition(position: _duration);
   },
   onChangeEnd: (double value) {
      audioProvider.invertSeekingState();
      audioProvider.audioSeek(seconds: value / 1000);
   },
   activeColor: primaryColor,
   inactiveColor: Colors.white.withOpacity(.2),
)
```

Slider.adaptive is a built-in slider that render a material slider on Android and a Cupertino slider on iOS. 
On Android I have no problems, everything just works fine. 

On iOS, where it is used Cupertino slider, I got this error message:

> Failed assertion: line 330 pos 15: 'value != null &&
value >= 0.0 && value <= 1.0': is not true.

(This assertion is not present in Material Slider)

Basically the assertion is raised by class "_RenderCupertinoSlider". 
As I've shown in the code above, I've set "min" and "max" parameter to define the range of values, which in this case are 0 and 80000, the duration of a song in milliseconds. 

**What I cannot understand is why having set "min" and "max" parameter, and in particular this second one greater than 1.0, is there a need to check if value is less than 1.0?** Obviously is not, I've indeed set "min" and "max" value for this purpose.

Anyone could help me and explain me that?

Thanks

**Comment by pedromassangocode** [comment]
Hi @LeLoMAu 
Can you please provide your `flutter doctor -v` and a minimal complete reproducible code sample? I need you to provide a code that I can just copy/paste to reproduce the issue.

Thank you

**Comment by LeLoMAu** [comment]
Here below the output of my `flutter doctor -v`:

> matteozantedeschi@MacBook-Pro ~ % /Users/matteozantedeschi/flutter/bin/flutter doctor -v
> [✓] Flutter (Channel stable, 1.20.4, on Mac OS X 10.15.6 19G2021, locale it-IT)
>     • Flutter version 1.20.4 at /Users/matteozantedeschi/flutter
>     • Framework revision fba99f6cf9 (3 settimane fa), 2020-09-14 15:32:52 -0700
>     • Engine revision d1bc06f032
>     • Dart version 2.9.2
> 
>  
> [✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
>     • Android SDK at /Users/matteozantedeschi/Library/Android/sdk
>     • Platform android-30, build-tools 30.0.0
>     • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
>     • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
>     • All Android licenses accepted.
> 
> [✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     • Xcode 12.0.1, Build version 12A7300
>     • CocoaPods version 1.10.0.rc.1
> 
> [✓] Android Studio (version 4.0)
>     • Android Studio at /Applications/Android Studio.app/Contents
>     • Flutter plugin version 50.0.1
>     • Dart plugin version 193.7547
>     • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
> 
> [!] VS Code (version 1.47.3)
>     • VS Code at /Applications/Visual Studio Code.app/Contents
>     ✗ Flutter extension not installed; install from
>       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
> 
>  
> [✓] Connected device (1 available)            
>     • iPhone 11 Pro (mobile) • 00E1E4BE-D794-4DF3-8046-4736E25BC2BE • ios • com.apple.CoreSimulator.SimRuntime.iOS-14-0
>       (simulator)
> 
> ! Doctor found issues in 1 category.

I will try to provide you a minimal complete reproducible code sample, because the one used by me is quite complex. 
But basically I think that to reproduce the issue just need something which invoke _RenderCupertinoSlider with a value greater than 1.0.

I noticed that this class (_RenderCupertinoSlider) is invoked only when I pop the current screen with slider and while the music is still playing (and conceptually the slider value still in progress) I go back to it. 
When I just remain on the screen _RenderCupertinoSlider is not invoked and therefore values greater than 1.0 are not a problem because the assertion is not evaluated.
